crates/eips/src/eip2718.rs

```diff
@@ -18,19 +18,51 @@ const TX_TYPE_BYTE_MAX: u8 = 0x7f;
 /// [EIP-2718] decoding errors.
 ///
 /// [EIP-2718]: https://eips.ethereum.org/EIPS/eip-2718
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug)]
 pub enum Eip2718Error {
     /// Rlp error from [`alloy_rlp`].
     RlpError(alloy_rlp::Error),
     /// Got an unexpected type flag while decoding.
     UnexpectedType(u8),
+    /// Missing `gasPrice` field for Legacy transaction.
+    MissingGasPrice,
+    /// Missing signature for transaction.
+    MissingSignature,
+    /// Missing `accessList` field for EIP-2930 transaction.
+    MissingAccessList,
+    /// Missing `maxFeePerGas` field for EIP-1559 transaction.
+    MissingMaxFeePerGas,
+    /// Missing `maxPriorityFeePerGas` field for EIP-1559 transaction.
+    MissingMaxPriorityFeePerGas,
+    /// Missing `maxFeePerBlobGas` field for EIP-4844 transaction.
+    MissingMaxFeePerBlobGas,
+    /// Missing `chainId` field for EIP-155 transaction.
+    MissingChainId,
+    /// Signature error from [`alloy_primitives`].
+    SignatureError(SignatureError),
```

Comment:
none of these are 2718 errors. they should not be in this enum

Reply:
What'd be the best place for them?